ee/app/livechat-enterprise/server/lib/LivechatEnterprise.js

```diff
@@ -161,3 +163,58 @@ export const LivechatEnterprise = {
 		updateRoomPriorityHistory(roomId, user, priority);
 	},
 };
+
+const RACE_TIMEOUT = 1000;
+const QUEUE_TIMEOUT = 5000;
+
+const queueWorker = {
+	running: false,
+	pool: [],
+	activeQueues: [],
+	start() {
+		if (this.running) {
+			return;
+		}
+
+		this.getActiveQueues();
+		OmnichannelQueue.initQueue();
+		this.running = true;
+		this.execute();
+	},
+	stop() {
+		this.running = false;
+		OmnichannelQueue.stopQueue();
+	},
+	async getActiveQueues() {
+		this.activeQueues = (await LivechatInquiry.getDistinctQueuedDepartments()).map(({ _id }) => _id);
+		setTimeout(this.getActiveQueues.bind(this), QUEUE_TIMEOUT);
+	},
+	nextQueue() {
+		if (!this.pool.length) {
+			this.pool = [...this.activeQueues];
```

Comment:
why are you not calling `getActiveQueues` here instead of having it being called every 5 seconds? I didn't understand the whole process after first look.. 😅 

I think making it clear that the pool is being populated only if it is empty might help understanding the code.

Reply:
Yeah, your suggestion is exactly how it was implemented :D 
The point is that I was thinking about a small number of different queues, then the method will be called more often but since we're now running it sequentially, there's no problem calling it this way.